### PR TITLE
AssemblyRegionWalker: add support for outputting assembly regions and activity profile in IGV format

### LINKED ISSUE #2796: Mutect2 and HaplotypeCaller lack the option to output active regions
This is a feature in GATK3 HaplotypeCaller that is yet to be ported to GATK4 HaplotypeCaller (which is in beta currently).

Given Mutect2 is ~~no longer~~ `still` in beta ~~(in production)~~, might consider adding this option.

The exact parameter name is `-ARO` or `--activeRegionOut`.

### PR DESCRIPTION
**Pull request by droazen**
This adds --activityProfileOut and --assemblyRegionOut arguments to AssemblyRegionWalker,
allowing assembly regions and supporting activity profile data to be output in a format
suitable for loading in IGV. Mirrors functionality present in GATK 3.x.

Resolves #2796

### CODE REVIEW

**Comment by droazen** [comment]
@jamesemery please review

File: src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java [hunk 1]
```diff
@@ -186,6 +221,37 @@ protected final void onStartup() {
...
+            catch ( IOException e ) {
+                throw new UserException.CouldNotCreateOutputFile(assemblyRegionOut, "Error writing assembly regions to output file", e);
+            }
+
+            logger.info("Writing assembly regions to " + assemblyRegionOut);
+            printIGVFormatHeader(assemblyRegionOutStream, "line", "AssemblyRegions");
+        }
+    }
+
+    private void printIGVFormatHeader(final PrintStream out, final String graphType, final String ... columns ) {
```
[the remark this answers is not in the document]
Comment (jamesemery) [reply src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java]: If you pull this out it makes more sense to leave that argument

File: src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java [hunk 2]
```diff
@@ -260,6 +327,34 @@ private void processReadShard(Shard<GATKRead> shard, ReferenceDataSource referen
...
+    private void writeActivityProfile(final List<ActivityProfileState> states) {
+        if ( activityProfileOutStream != null ) {
+            for ( final ActivityProfileState state : states ) {
+                printIGVFormatRow(activityProfileOutStream, state.getLoc(), "state", Math.min(state.isActiveProb(), 1.0));
+            }
+        }
+    }
+
+    private void printIGVFormatRow(final PrintStream out, final Locatable loc, final String featureName, final double ... values) {
+        // note that start and stop are 0 based, but the stop is exclusive so we don't subtract 1
```
[the remark this answers is not in the document]
Comment (droazen) [reply src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java]: I've moved these methods to a new `IGVUtils` class, and added direct unit tests and javadoc for them.

File: src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (droazen) [reply src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java]: Resolved by extracting to a separate class and adding docs.

**Comment by droazen** [comment]
Comments addressed, back to @jamesemery for a last look.

**Pull request merged**